.github/workflows/markdown-link-check.yml

```diff
@@ -1,7 +1,14 @@
 name: Check Markdown links
 on:
   push:
+    branches: [ main ]
```

Comment:
What was your motivation for restricting to `main`? I would think we would want to run this checker *before* merging to main.

Reply:
I copied this part from the codeql checks, which seem to run on every push to a PR against main (see below check).

I suppose this one doesn't matter because main is protected, but it does mean that we won't have github actions run on people's dev branches if they aren't part of a PR yet.

Reply:
Push to `main` doesn't make sense in our repos since pushing to `main` is blocked. You should just use the `pull_request` action.